rados/snapshot.go

```diff
@@ -70,3 +75,77 @@ func (snapshot *Snapshot) Remove() error {
 	ret := C.rados_ioctx_snap_remove(snapshot.ioctx.ioctx, cSnapName)
 	return getError(ret)
 }
+
+// SnapID represents the ID of a rados snapshot.
+type SnapID C.rados_snap_t
+
+// LookupSnapshot gets the ID of a pool snapshot.
+//
+// Implements:
+//  int rados_ioctx_snap_lookup(rados_ioctx_t io, const char *name, rados_snap_t *id)
+func (snapshot *Snapshot) LookupSnapshot() (SnapID, error) {
+	var snapID SnapID
+
+	if err := snapshot.validate(); err != nil {
+		return snapID, err
+	}
+
+	cSnapName := C.CString(snapshot.name)
+	defer C.free(unsafe.Pointer(cSnapName))
+
+	ret := C.rados_ioctx_snap_lookup(
+		snapshot.ioctx.ioctx,
+		cSnapName,
+		(*C.rados_snap_t)(&snapID))
+	return snapID, getError(ret)
+}
+
+// GetSnapName fetches the name of a pool snapshot with the given snapshot ID.
+//
+// Implements:
+//	int rados_ioctx_snap_get_name(rados_ioctx_t io, rados_snap_t id, char *name, int maxlen)
+func (ioctx *IOContext) GetSnapName(snapID SnapID) (string, error) {
```

Comment:
How about calling this function `GetSnapByID()` and returning a Snapshot object?

Maybe the Snapshot type should have `GetName()` and `GetID()` methods too?

Reply:
>> How about calling this function GetSnapByID() and returning a Snapshot object?
Yeah, that can be done.

>> Maybe the Snapshot type should have GetName() and GetID() methods too?
Lookup() is equivalent to GetID() method, GetName() can be done but I feel its kind of duplicate work with GetSnapByID
Comments?